test/jdk/java/net/httpclient/ExpectContinueTest.java

```diff
@@ -0,0 +1,270 @@
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @summary Tests that when the httpclient sends a 100 Expect Continue header and receives
+ *          a response code of 417 Expectation Failed, that the client does not hang
+ *          indefinitely and closes the connection.
+ * @bug 8286171
+ * @modules java.base/sun.net.www.http
+ *          java.net.http/jdk.internal.net.http.common
+ *          java.net.http/jdk.internal.net.http.frame
+ *          java.net.http/jdk.internal.net.http.hpack
+ *          jdk.httpserver
+ * @library /test/lib http2/server
+ * @compile HttpServerAdapters.java
+ * @run testng/othervm/timeout=10  ExpectContinueTest
+ */
+
+
+import com.sun.net.httpserver.HttpServer;
+
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import javax.net.ServerSocketFactory;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.io.PrintWriter;
+import java.io.Writer;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.StringTokenizer;
+import java.util.concurrent.CompletableFuture;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+public class ExpectContinueTest implements HttpServerAdapters {
+
+    HttpServerAdapters.HttpTestServer httpTestServer;    // HTTP/1.1
+    HttpServerAdapters.HttpTestServer http2TestServer;   // HTTP/2
+    Http1HangServer http1HangServer;
+
+    URI getUri;
+    URI postUri;
+    URI hangUri;
+    URI h2getUri;
+    URI h2postUri;
+    URI h2hangUri;
+
+    @BeforeTest
+    public void setup() throws Exception {
+        InetSocketAddress sa = new InetSocketAddress(InetAddress.getLoopbackAddress(), 0);
+        InetSocketAddress saHang = new InetSocketAddress(InetAddress.getLoopbackAddress(), 0);
+
+        httpTestServer = HttpServerAdapters.HttpTestServer.of(HttpServer.create(sa, 0));
```

Comment:
You should be able to write:
```
httpTestServer = HttpTestServer.of(HttpServer.create(sa, 0));
```
here. No need for the `HttpServerAdapters` prefix.